Cargo.toml

```diff
@@ -287,6 +287,7 @@ rsa_signing = []
 slow_tests = []
 test_logging = []
 use_heap = []
+bn_tests = []
```

Comment:
The way this ended up, `bn_tests` won't be enabled unless somebody specifically asks for it. That's not what we want, right?

I suggest that we do this instead:
1. Make the `rsa_signing` feature depend on the `bn_tests` feature. That way, whenever RSA signing is enabled, the `bn_tests` will be built and run. But when RSA signing isn't enabled, the `bn_tests` will be skipped.

2. You can see that we already have disabled bn_tests for macOS/iOS targets for similar reasons (trouble with C++) even when RSA signing is enabled. I suggest we add `target_env = "musl"` to the list of conditions where the bn_tests are skipped.

WDYT?